### PR TITLE
Clarify some comments on RateLimiterOptionsExtensions

### PR DESCRIPTION
**Pull request by wtgodbe**
- Clarify what the extension methods do & how to use them
- Clarify that `AutoReplenishment` doesn't actually do anything

### CODE REVIEW

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 1]
```diff
@@ -34,12 +41,18 @@ public static RateLimiterOptions AddTokenBucketLimiter(this RateLimiterOptions o
     }
 
     /// <summary>
-    /// Adds a new <see cref="FixedWindowRateLimiter"/> with the given <see cref="FixedWindowRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// Registers a new <see cref="FixedWindowRateLimiter"/> with the given <see cref="FixedWindowRateLimiterOptions"/> to the to the application.
```
Comment (sebastienros) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:44]: I would have like to see something like

> Adds a limit to the number of requests that can be executed in a specific time window.

Then for the sliding:

> Adds a limit to the number of requests that can be executed in a sliding time window.

This kind of help message. We should check with others if this is not the kind of message we want in API docs.

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 2]
```diff
@@ -11,12 +12,18 @@ namespace Microsoft.AspNetCore.RateLimiting;
 public static class RateLimiterOptionsExtensions
 {
     /// <summary>
-    /// Adds a new <see cref="TokenBucketRateLimiter"/> with the given <see cref="TokenBucketRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// Registers a new <see cref="TokenBucketRateLimiter"/> with the given <see cref="TokenBucketRateLimiterOptions"/> to the application.
+    /// The limiter can be added to an endpoint via <see cref="RateLimiterEndpointConventionBuilderExtensions.RequireRateLimiting{TBuilder}(TBuilder, string)"/>,
+    /// Or via the <see cref="EnableRateLimitingAttribute"/>.
```
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:17]:
```suggestion
    /// or via the <see cref="EnableRateLimitingAttribute"/>.
```

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs (hunk 1 above)
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:44]:
```suggestion
    /// Registers a new <see cref="FixedWindowRateLimiter"/> with the given <see cref="FixedWindowRateLimiterOptions"/> to the application.
```

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 3]
```diff
@@ -34,12 +41,18 @@ public static RateLimiterOptions AddTokenBucketLimiter(this RateLimiterOptions o
     }
 
     /// <summary>
-    /// Adds a new <see cref="FixedWindowRateLimiter"/> with the given <see cref="FixedWindowRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// Registers a new <see cref="FixedWindowRateLimiter"/> with the given <see cref="FixedWindowRateLimiterOptions"/> to the to the application.
+    /// The limiter can be added to an endpoint via <see cref="RateLimiterEndpointConventionBuilderExtensions.RequireRateLimiting{TBuilder}(TBuilder, string)"/>,
+    /// Or via the <see cref="EnableRateLimitingAttribute"/>.
```
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:46]:
```suggestion
    /// or via the <see cref="EnableRateLimitingAttribute"/>.
```

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 4]
```diff
@@ -57,12 +70,18 @@ public static RateLimiterOptions AddFixedWindowLimiter(this RateLimiterOptions o
     }
 
     /// <summary>
-    /// Adds a new <see cref="SlidingWindowRateLimiter"/> with the given <see cref="SlidingWindowRateLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// Registers a new <see cref="SlidingWindowRateLimiter"/> with the given <see cref="SlidingWindowRateLimiterOptions"/> to the to the application.
```
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:73]:
```suggestion
    /// Registers a new <see cref="SlidingWindowRateLimiter"/> with the given <see cref="SlidingWindowRateLimiterOptions"/> to the application.
```

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 5]
```diff
@@ -57,12 +70,18 @@ public static RateLimiterOptions AddFixedWindowLimiter(this RateLimiterOptions o
...
+    /// The limiter can be added to an endpoint via <see cref="RateLimiterEndpointConventionBuilderExtensions.RequireRateLimiting{TBuilder}(TBuilder, string)"/>,
+    /// Or via the <see cref="EnableRateLimitingAttribute"/>.
```
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:75]:
```suggestion
    /// or via the <see cref="EnableRateLimitingAttribute"/>.
```

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs (hunk 2 above)
Comment (BrennanConroy) [reply src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:17]: Maybe this should be "limiter can be **applied** to an endpoint ..."

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 6]
```diff
@@ -80,7 +99,9 @@ public static RateLimiterOptions AddSlidingWindowLimiter(this RateLimiterOptions
     }
 
     /// <summary>
-    /// Adds a new <see cref="ConcurrencyLimiter"/> with the given <see cref="ConcurrencyLimiterOptions"/> to the <see cref="RateLimiterOptions"/>.
+    /// Registers a new <see cref="ConcurrencyLimiter"/> with the given <see cref="ConcurrencyLimiterOptions"/> to the to the application.
```
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:102]:
```suggestion
    /// Registers a new <see cref="ConcurrencyLimiter"/> with the given <see cref="ConcurrencyLimiterOptions"/> to the application.
```

File: src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs [hunk 7]
```diff
@@ -80,7 +99,9 @@ public static RateLimiterOptions AddSlidingWindowLimiter(this RateLimiterOptions
...
+    /// The limiter can be added to an endpoint via <see cref="RateLimiterEndpointConventionBuilderExtensions.RequireRateLimiting{TBuilder}(TBuilder, string)"/>,
+    /// Or via the <see cref="EnableRateLimitingAttribute"/>.
```
Comment (BrennanConroy) [inline src/Middleware/RateLimiting/src/RateLimiterOptionsExtensions.cs:104]:
```suggestion
    /// or via the <see cref="EnableRateLimitingAttribute"/>.
```

**Comment by wtgodbe** [comment]
@mkArtakMSFT looks like we need to give the bot extra permissions - should we add it to https://github.com/orgs/dotnet/teams/npt?

**Review by adityamandaleeka — APPROVED** [review]

**Comment by adityamandaleeka** [comment]
@wtgodbe Is this ready to merge?

**Comment by wtgodbe** [comment]
> @wtgodbe Is this ready to merge?

Go for it - @sebastienros had some requests but those can be addressed in a separate PR if necessary

**Pull request merged**